File: test-functions/ruby/lambda_function.rb
require 'json'

def lambda_handler(event:, context:)
  test_gems = event['test_imports'] || ['nokogiri']
  results = {}
  test_gems.each do |gem_name|
    begin
      require gem_name
      mod = Object.const_get(gem_name.split('-').map(&:capitalize).join)
      version = mod.const_defined?(:VERSION) ? mod::VERSION : 'unknown'
      results[gem_name] = { status: 'ok', version: version }
    rescue LoadError => e
      results[gem_name] = { status: 'error', message: e.message }
    rescue NameError
      results[gem_name] = { status: 'ok', version: 'loaded' }
    end
  end
  {
    statusCode: 200,
    body: JSON.generate(results)
  }
end

File: test-functions/python/lambda_function.py
import json


def lambda_handler(event, context):
    """Test that layer dependencies are importable."""
    results = {}
    test_imports = event.get('test_imports', ['requests'])
    for module_name in test_imports:
        try:
            mod = __import__(module_name)
            version = getattr(mod, '__version__', 'unknown')
            results[module_name] = {'status': 'ok', 'version': version}
        except ImportError as e:
            results[module_name] = {'status': 'error', 'message': str(e)}
    return {
        'statusCode': 200,
        'body': json.dumps(results, indent=2),
    }
